unlock-app/src/components/interface/connect/Wallet.tsx

```diff
@@ -29,14 +28,16 @@ export const ConnectWallet = ({ onUnlockAccount }: ConnectWalletProps) => {
   return (
     <div className="space-y-6 divide-y divide-gray-100">
       <div className="grid gap-4 px-6">
-        <ConnectButton
-          icon={<SvgComponents.Metamask width={40} height={40} />}
-          highlight={recentlyUsedProvider === 'METAMASK'}
-          loading={isConnecting === 'METAMASK'}
-          onClick={createOnConnectHandler('METAMASK')}
-        >
-          Metamask
-        </ConnectButton>
+        {window.ethereum && (
+          <ConnectButton
+            icon={<SvgComponents.Metamask width={40} height={40} />}
+            highlight={recentlyUsedProvider === 'METAMASK'}
+            loading={isConnecting === 'METAMASK'}
+            onClick={createOnConnectHandler('METAMASK')}
+          >
+            Metamask
+          </ConnectButton>
+        )}
```

Comment:
I am not sure this works on mobile? We may still want to show the MetaMask button?